### ISSUE TITLE
add a matchesSelector() and doesNotMatchSelector() assertion

### ISSUE THREAD

**Issue by timo-lindemann-victors** [simplabs/qunit-dom#206]
Would it be a good idea to have something like

```js
assert.dom(selector).matchesSelector(anotherSelector, message)
```

to be able to check if a given CSS selector returns the exact same (set of?) element(s)?

I would probably save some boilerplate if we had that, and CSS selectors have come to express a lot more things than just CSS classes and attributes (sibling detection, last-child, ...)

Just checking the waters here.

**Comment by Turbo87** [comment]
sound like a good idea 👍 

two more thoughts:

1. the snippet above uses `matches()` but as the issue title suggests we should use `matchesSelector()` to be a little more specific

2. if we have an `<div id="foo" class="red">` and other uses of `red` on the page, what would `assert.dom('#foo').matchesSelector('.red')` result in?

**Comment by timo-lindemann-victors** [comment]
@Turbo87 glad you like it.

 Thanks for pointing out my mistake; I have corrected the name of the assertion in my opening question.

To your question: The assertion should use [Element.matches](https://developer.mozilla.org/en-US/docs/Web/API/Element/matches) so the answer is most likely a passing assertion for `matchesSelector ` and a failure for `doesNotMatchSelector`.

So, if the `anotherSelector` returns a set of elements containing the element of the first selector, the assertion passes.

I have not decided what to do if `selector` describes multiple elements. I think the most sensible approach is to check that the result set of `selector` is s subset of `anotherSelector`.

Thoughts?

**Comment by lindem** [comment]
I might play around with it on sunday.

**Comment by timo-lindemann-victors** [comment]
Fixed by #208.
